server/etcdmain/util.go

```diff
@@ -20,11 +20,14 @@ import (
 
 	"go.etcd.io/etcd/client/pkg/v3/srv"
 	"go.etcd.io/etcd/client/pkg/v3/transport"
+	"go.etcd.io/etcd/client/pkg/v3/verify"
 
 	"go.uber.org/zap"
 )
 
 func discoverEndpoints(lg *zap.Logger, dns string, ca string, insecure bool, serviceName string) (s srv.SRVClients) {
+	verify.Assert(lg != nil, "nil log isn't allowed")
```

Comment:
Just as I mentioned, we should only add assertion like this in user-facing (exported) methods/functions. For internal functions & methods, we have 100% control over the usage, why should we add such assertion? 

```suggestion
```

Reply:
Indeed, I've investigated and no non-empty lg is passed in, just removing the validation is enough

Reply:
thank you